### PR TITLE
website/integrations: add FreshRSS

### PR DESCRIPTION
**Pull request by foux**
## Details

This PR adds community documentation to login to FreshRSS with Authentik

---

-   [X] The documentation has been updated
-   [X] The documentation has been formatted (`make website`)

### CODE REVIEW

**Comment by XtremeOwnageDotCom** [comment]
As a note- I noticed this line-

```
:::info
This integration only works with the Docker install of FreshRSS, using FreshRSS docker image, on x86_64 systems and with
the Debian version of the image.
```

I can confirm, it will work with kubernetes too. Which- is where I based my version of the documentation from. Just- make sure to use the FreshRSS/FreshRSS repo, and not linuxserver/FreshRSS

**Comment by Alkarex** [comment]
You could consider adding a link to
https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect-Authentik.html
or
https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect.html
depending on what makes more sense

**Comment by foux** [comment]
> You could consider adding a link to https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect-Authentik.html or https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect.html depending on what makes more sense

Done. I've added link to the FreshRSS documentation. You can see what the doc will look like here : https://deploy-preview-7301--authentik.netlify.app/integrations/services/freshrss/

File: website/sidebarsIntegrations.js [hunk 1]
```diff
@@ -97,6 +97,7 @@ module.exports = {
                     label: "Miscellaneous",
                     items: [
                         "services/gravitee/index",
+                        "services/freshrss/index",
```
Comment (rissson) [inline website/sidebarsIntegrations.js:100]:
```suggestion
                        "services/freshrss/index",
                        "services/gravitee/index",
```
We're trying to keep this list sorted.

File: website/integrations/services/freshrss/index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+> FreshRSS is a self-hosted RSS feed aggregator.
+>
+> -- https://github.com/FreshRSS/FreshRSS
+
+## Preparation
+
+The following placeholders will be used:
+
+-   `service.company` is the FQDN of the FreshRSS install.
```
Comment (rissson) [inline website/integrations/services/freshrss/index.md:17]:
```suggestion
-   `freshrss.company` is the FQDN of the FreshRSS install.
```
And the same for the rest of the document.

File: website/integrations/services/freshrss/index.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+_Client Type_ : _Confidential_
+
+:::note
+Take note of the `Client ID` and `Client Secret`, you'll need them later.
+:::
+
+_Redirect URIs/Origins_ :
+
+-   `https://service.company/i/oidc/`
+-   `https://service.company:port/i/oidc`
```
Comment (rissson) [inline website/integrations/services/freshrss/index.md:35]: Can we assume in this document that freshrss is running behing a reverse proxy and thus that `:port` is not required in the URL? (We can then also remove it from the above placeholders)

File: website/integrations/services/freshrss/index.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+Then click _Finish_ to create your provider.
+
+Then create an _Application_, note its slug, and assign it to the provider you've just created.
+
+## FreshRSS Configuration
+
+:::info
+This integration only works with the Docker or Kubernetes install of FreshRSS, using [FreshRSS docker image](https://hub.docker.com/r/freshrss/freshrss/), on x86_64 systems and with the Debian version of the image.
+:::
```
Comment (rissson) [inline website/integrations/services/freshrss/index.md:47]: Do you mind giving a bit more context about this?

File: website/integrations/services/freshrss/index.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Add those environment variables to your _Docker_ image :
+
+-   `OIDC_ENABLED` : `1`
+-   `OIDC_PROVIDER_METADATA_URL` : `https://authentik.company/application/o/slug/.well-known/openid-configuration` replacing slug with the slug of your created application
```
Comment (rissson) [inline website/integrations/services/freshrss/index.md:52]:
```suggestion
-   `OIDC_PROVIDER_METADATA_URL` : `https://authentik.company/application/o/<application-slug>/.well-known/openid-configuration` replacing `<application-slug>` with the slug of your created application
```

File: website/integrations/services/freshrss/index.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+-   `OIDC_CLIENT_ID` : the client ID of your provider
+-   `OIDC_CLIENT_SECRET` : the client secret of your provider
+-   `OIDC_X_FORWARDED_HEADERS` : `X-Forwarded-Port X-Forwarded-Proto X-Forwarded-Host`
```
Comment (rissson) [inline website/integrations/services/freshrss/index.md:55]: This seems odd. That should be a reverse proxy configuration.

File: website/integrations/services/freshrss/index.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+-   `OIDC_SCOPES` : `openid email profile`
+
+Before restarting your Docker container, ensure that one of the Admin users of your FreshRSS instance has the same login as one of your Authentik user.
+
+Restart your FreshRSS container, and login as a user that exists on both FreshRSS and your Authentik.
+Navigate to _Settings_ > _Authentication_ in your FreshRSS instance, and choose as an authentication method _HTTP (for advanced users with HTTPS)_
+
+You can find additional informations on [FreshRSS documentation](https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect.html)
```
Comment (rissson) [inline website/integrations/services/freshrss/index.md:63]:
```suggestion
You can find additional information on [FreshRSS documentation](https://freshrss.github.io/FreshRSS/en/admins/16_OpenID-Connect.html)
```

**Review by rissson — CHANGES_REQUESTED** [review]
Hi! Thank you for your contribution! A few nitpicks here and there to align a bit more with what's done for the rest of the documentation, but looks very good overall.
I'm a bit unsure about some of the stuff about the reverse proxy, I'll need to look into the freshrss documentation a bit more.

File: website/integrations/services/freshrss/index.md (hunk 3 above)
Comment (foux) [reply website/integrations/services/freshrss/index.md:35]: Hello. @rissson. Unfortunately we can't, we need to add the :port alternative, even behind a reverse proxy, that's why I mentioned it

File: website/integrations/services/freshrss/index.md (hunk 4 above)
Comment (foux) [reply website/integrations/services/freshrss/index.md:47]: This integration works with Apache OIDC extensions. On the other images (specifically linuxserver one), those extension aren't present.

They are only here on the debian version of the image on x86_64 platform.

You can have more informations on this issue : https://github.com/FreshRSS/FreshRSS/issues/5722

File: website/sidebarsIntegrations.js (hunk 1 above)
Comment (foux) [reply website/sidebarsIntegrations.js]: Done. Sorry about that, I was looking for alphabetical order, and missed the first one....

File: website/integrations/services/freshrss/index.md (hunk 2 above)
Comment (foux) [reply website/integrations/services/freshrss/index.md]: Done, thanks!

File: website/integrations/services/freshrss/index.md (hunk 5 above)
Comment (foux) [reply website/integrations/services/freshrss/index.md]: Done, thanks

File: website/integrations/services/freshrss/index.md (hunk 7 above)
Comment (foux) [reply website/integrations/services/freshrss/index.md:63]: Done, thanks

File: website/integrations/services/freshrss/index.md (hunk 6 above)
Comment (foux) [reply website/integrations/services/freshrss/index.md:55]: > This seems odd. That should be a reverse proxy configuration.

What do you mean? About the `OIDC_X_FORWARDED_HEADERS`? Apache OIDC extension needs them behind the reverse proxy

File: website/integrations/services/freshrss/index.md (hunk 4 above)
Comment (foux) [reply website/integrations/services/freshrss/index.md]: Done

Comment (XtremeOwnageDotCom) [reply website/integrations/services/freshrss/index.md]: https://github.com/FreshRSS/FreshRSS/pull/5764

Lol, might not be an issue at all, coming soon.

Comment (Alkarex) [reply website/integrations/services/freshrss/index.md]: Still now working fully with Alpine (and the package is still in "testing"), and still not on ARM

**Comment by foux** [comment]
@rissson is everything good for you?

File: website/integrations/services/freshrss/index.md (hunk 6 above)
Comment (rissson) [reply website/integrations/services/freshrss/index.md:55]: I see. We're linking to the FreshRSS documentation which has all the relevant details, so fine by me.

File: website/integrations/services/freshrss/index.md (hunk 4 above)
Comment (foux) [reply website/integrations/services/freshrss/index.md]: Will do!

Comment (foux) [reply website/integrations/services/freshrss/index.md]: Done

**Review by rissson — APPROVED** [review]

**Comment by foux** [comment]
@rissson  can we please relauch the checks?
Thanks

**Pull request merged**